pkg/cloudprovider/providers/gce/cloud/gen.go

```diff
@@ -0,0 +1,10446 @@
+	return mock.MockHttpHealthChecks
+}
+
+// HttpsHealthChecks returns the interface for the ga HttpsHealthChecks.
+func (mock *MockGCE) HttpsHealthChecks() HttpsHealthChecks {
+	return mock.MockHttpsHealthChecks
+}
+
+// InstanceGroups returns the interface for the ga InstanceGroups.
+func (mock *MockGCE) InstanceGroups() InstanceGroups {
+	return mock.MockInstanceGroups
+}
+
+// Instances returns the interface for the ga Instances.
+func (mock *MockGCE) Instances() Instances {
+	return mock.MockInstances
+}
+
+// BetaInstances returns the interface for the beta Instances.
+func (mock *MockGCE) BetaInstances() BetaInstances {
+	return mock.MockBetaInstances
+}
+
+// AlphaInstances returns the interface for the alpha Instances.
+func (mock *MockGCE) AlphaInstances() AlphaInstances {
+	return mock.MockAlphaInstances
+}
+
+// AlphaNetworkEndpointGroups returns the interface for the alpha NetworkEndpointGroups.
+func (mock *MockGCE) AlphaNetworkEndpointGroups() AlphaNetworkEndpointGroups {
+	return mock.MockAlphaNetworkEndpointGroups
+}
+
+// Projects returns the interface for the ga Projects.
+func (mock *MockGCE) Projects() Projects {
+	return mock.MockProjects
+}
+
+// Regions returns the interface for the ga Regions.
+func (mock *MockGCE) Regions() Regions {
+	return mock.MockRegions
+}
+
+// Routes returns the interface for the ga Routes.
+func (mock *MockGCE) Routes() Routes {
+	return mock.MockRoutes
+}
+
+// SslCertificates returns the interface for the ga SslCertificates.
+func (mock *MockGCE) SslCertificates() SslCertificates {
+	return mock.MockSslCertificates
+}
+
+// TargetHttpProxies returns the interface for the ga TargetHttpProxies.
+func (mock *MockGCE) TargetHttpProxies() TargetHttpProxies {
+	return mock.MockTargetHttpProxies
+}
+
+// TargetHttpsProxies returns the interface for the ga TargetHttpsProxies.
+func (mock *MockGCE) TargetHttpsProxies() TargetHttpsProxies {
+	return mock.MockTargetHttpsProxies
+}
+
+// TargetPools returns the interface for the ga TargetPools.
+func (mock *MockGCE) TargetPools() TargetPools {
+	return mock.MockTargetPools
+}
+
+// UrlMaps returns the interface for the ga UrlMaps.
+func (mock *MockGCE) UrlMaps() UrlMaps {
+	return mock.MockUrlMaps
+}
+
+// Zones returns the interface for the ga Zones.
+func (mock *MockGCE) Zones() Zones {
+	return mock.MockZones
+}
+
+// MockAddressesObj is used to store the various object versions in the shared
+// map of mocked objects. This allows for multiple API versions to co-exist and
+// share the same "view" of the objects in the backend.
+type MockAddressesObj struct {
+	Obj interface{}
+}
+
+// ToAlpha retrieves the given version of the object.
+func (m *MockAddressesObj) ToAlpha() *alpha.Address {
+	if ret, ok := m.Obj.(*alpha.Address); ok {
+		return ret
+	}
+	// Convert the object via JSON copying to the type that was requested.
+	ret := &alpha.Address{}
+	if err := copyViaJSON(ret, m.Obj); err != nil {
```

Comment:
Would be nice to use Converter.Convert for this. However we may want to do that in a follow up PR.